src/random.cpp

```diff
@@ -34,23 +34,33 @@ POSSIBILITY OF SUCH DAMAGE.
 #include "libtorrent/random.hpp"
 #include "libtorrent/assert.hpp"
 
+#ifdef BOOST_NO_CXX11_HDR_RANDOM
```

Comment:
I tried to do it with `__cplusplus` but msvc 2015 still has a value of `199711L` for it!!!

For your worries about boost.config being included. This file at the top includes libtorrent/config.hpp. And that header includes boost/config.hpp at the top.
Or we could include boost/config.hpp here too just to be safe from future changes in libtorrent/config.hpp